pkg/tfvars/openstack/openstack.go

```diff
@@ -182,150 +122,130 @@ func tfVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, workerConfigs []*v1
 	// created by the installer from the URL. Otherwise, it means that we are given the name of the pre-created
 	// Glance image, which we should use for instances.
 	imageName, isURL := rhcos.GenerateOpenStackImageName(baseImage, infraID)
-	cfg.BaseImageName = imageName
 	if isURL {
 		// Valid URL -> use baseImage as a URL that will be used to create new Glance image with name "<infraID>-rhcos".
-		var localFilePath string
-
-		url, err := url.Parse(baseImage)
-		if err != nil {
-			return nil, err
-		}
-
-		// We support 'http(s)' and 'file' schemes. If the scheme is http(s), then we will upload a file from that
-		// location. Otherwise will take local file path from the URL.
-		switch url.Scheme {
-		case "http", "https":
-			localFilePath, err = cache.DownloadImageFile(baseImage)
-			if err != nil {
-				return nil, err
-			}
-		case "file":
-			localFilePath = filepath.FromSlash(url.Path)
-		default:
-			return nil, errors.Errorf("Unsupported URL scheme: '%v'", url.Scheme)
-		}
-
-		err = uploadBaseImage(cloud, localFilePath, imageName, infraID, baseImageProperties)
-		if err != nil {
+		if err := uploadBaseImage(cloud, baseImage, imageName, infraID, installConfig.Config.Platform.OpenStack.ClusterOSImageProperties); err != nil {
 			return nil, err
 		}
 	}
 
-	clientConfigCloud, err := clientconfig.GetCloudFromYAML(openstackdefaults.DefaultClientOpts(cloud))
-	if err != nil {
-		return nil, err
-	}
-
-	glancePublicURL, err := getGlancePublicURL(serviceCatalog, clientConfigCloud.RegionName)
-	if err != nil {
-		return nil, err
-	}
-
-	configLocation, err := uploadBootstrapConfig(cloud, bootstrapIgn, infraID)
+	serviceCatalog, err := getServiceCatalog(cloud)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not retrieve service catalog: %w", err)
 	}
 
-	tokenID, err := getAuthToken(cloud)
-	if err != nil {
-		return nil, err
+	var regionName string
```

Comment:
You could potentially move this into getBootstrapShim() as it doesn't appear to be used anywhere else. This way, that's one less arg to pass to getBootstrapShim().

Reply:
Good idea!